jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushImageStep.java

```diff
@@ -79,7 +80,10 @@
       DescriptorDigest manifestDigest = Digests.computeJsonDigest(manifestTemplate);
 
       Set<String> imageQualifiers =
-          singlePlatform ? tags : Collections.singleton(manifestDigest.toString());
+          singlePlatform
+              ? tags
+              : getChildTags(builtImage, tags, manifestDigest, buildContext.isEnablePlatformTags());
```

Comment:
Maybe we can put the entire `singlePlatform ? tags : ...` logic into `getChildTags()`. And not sure I like the word "child." Maybe just `getImageQualifiers()`.  And actually I think it's better to move this computation to line 61 so that it looks like

```diff
     Set<String> tags = buildContext.getAllTargetImageTags();
-    int numPushers = singlePlatform ? tags.size() : 1;
+    Set<String> imageQualifiers = getImageQualifiers(...);
...
-            progressEventDispatcherFactory.create("launching manifest pushers", numPushers)) {
+            progressEventDispatcherFactory.create("launching manifest pushers", imageQualifiers.size())) {
```

Reply:
Sure. it drags a few more locals up, not a big deal